@xen-orchestra/proxy/src/app/_disposeResourceOnStop.js

```diff
@@ -0,0 +1,18 @@
+import Resource from 'promise-toolbox/_Resource'
+
+export function disposeResourceOnStop(res, [getHooks], args) {
+  const hooks = getHooks.apply(this, args)
+
+  const dispose = value => {
+    if (res.d !== undefined) {
+      const d = res.d
+      res.d = undefined
+      return d(value)
+    }
+  }
+  hooks.on('stop', dispose)
+  return new Resource(res.p, value => {
+    hooks.removeListener('stop', dispose)
```

Comment:
Why don't we remove the listener when called by the event?

Reply:
`xo-proxy` will not stop if a listener is open?

Reply:
No, but why leave a listener if it does not do anything?

Also, after a while there might be hundreds or thousands of useless listeners there.